core/include/detray/core/detector.hpp

```diff
@@ -273,47 +292,75 @@ class detector {
     DETRAY_HOST_DEVICE
     inline auto surface_store() -> surface_container & { return _surfaces; }
 
-    /// @returns all surfaces - const
+    /// @returns all portals - const
+    /// @note Depending on the detector type, this can also contain other
+    /// surfaces
     DETRAY_HOST_DEVICE
-    inline const auto &surfaces() const {
+    inline const auto &portals() const {
```

Comment:
Not all surfaces can be found in the brute force method anymore, as this now handles only the portals. Therefore, a comprehensive surface lookup was added to the detector. This corresponds to the former global surface vector